File: Kotlin_DB/app/src/main/java/io/github/karan/kotlin_db/ModifyEmployeeActivity.java
package io.github.karan.kotlin_db;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;

public class ModifyEmployeeActivity extends AppCompatActivity {

    private EditText nameText;
    private EditText addressText;

    private long _id;

    private DatabaseHelper myHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_modify_employee);

        myHelper = new DatabaseHelper(this);
        myHelper.open();

        nameText = findViewById(R.id.name_edittext);
        addressText = findViewById(R.id.address_edittext);

        Intent intent = getIntent();
        String id = intent.getStringExtra("id");
        String name = intent.getStringExtra("name");
        String desc = intent.getStringExtra("address");

        _id = Long.parseLong(id);

        nameText.setText(name);
        addressText.setText(desc);
    }

    private void returnToMainActivity()
    {
        finish();
    }

    public void updateButtonPressed(View view) {
        String name = nameText.getText().toString();
        String address = addressText.getText().toString();

        myHelper.update(_id, name, address);
        returnToMainActivity();
    }

    public void deleteButtonPressed(View view) {
        myHelper.delete(_id);
        returnToMainActivity();

    }
}

File: Kotlin_DB/app/src/main/java/io/github/karan/kotlin_db/ModifyEmployeeActivity.kt
package io.github.karan.kotlin_db

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.EditText

class ModifyEmployeeActivity : AppCompatActivity() {

    private var nameText: EditText? = null
    private var addressText: EditText? = null

    private var _id: Long = 0

    private var myHelper: DatabaseHelper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_modify_employee)

        myHelper = DatabaseHelper(this)
        myHelper!!.open()

        nameText = findViewById(R.id.name_edittext)
        addressText = findViewById(R.id.address_edittext)

        val intent = intent
        val id = intent.getStringExtra("id")
        val name = intent.getStringExtra("name")
        val desc = intent.getStringExtra("address")

        _id = java.lang.Long.parseLong(id)

        nameText!!.setText(name)
        addressText!!.setText(desc)
    }

    private fun returnToMainActivity() {
        finish()
    }

    fun updateButtonPressed(view: View) {
        val name = nameText!!.text.toString()
        val address = addressText!!.text.toString()

        myHelper!!.update(_id, name, address)
        returnToMainActivity()
    }

    fun deleteButtonPressed(view: View) {
        myHelper!!.delete(_id)
        returnToMainActivity()

    }
}

